task8-rashid-ab/src/Producer.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Producer.h"
+#include <iostream>
+
+std::mutex mtx;
+std::condition_variable cv;
+bool finished = false;
```

Comment:
Is this static? Is is shared between the different threads? This is very unclear..